src/python/UserFileCache/RESTExtensions.py

```diff
@@ -11,33 +11,48 @@
 from WMCore.REST.Validation import _validate_one
 from WMCore.REST.Error import RESTError, InvalidParameter
 from WMCore.Services.UserFileCache.UserFileCache import calculateChecksum
+from WMCore.REST.Auth import get_user_info
 
 # external dependecies here
 import tarfile
 import hashlib
 import cStringIO
-import cherrypy
+import urllib2
 from os import fstat, walk, path, listdir
 
 # 600MB is the default user quota limit - overwritten in RESTBaseAPI if quota_user_limit is set in the config
 QUOTA_USER_LIMIT = 1024*1024*600
 #these users have 10* basic user quota - overwritten in RESTBaseAPI if powerusers is set in the config
 POWER_USERS_LIST = []
 
+def http_error(msg, code=403):
+    try:
+        import requests
+        err = requests.HTTPError(msg)
+        err.response.status_code = code
+        return err
+    except ImportError:
+        url = '' # required for urllib2 HTTPError but we should acquire it from elsewhere
```

Comment:
@vkuznet is this comment meaning that I should do something about aquiring this from somewhere ?

Reply:
Well, the new way to do is to use requests python library which takes care of everything, but it is not standard python library (i.e. 3rd party product). The standard python library is urllib2 which provides the HTTPError, but it has 6 attributes to fill in. One of them is url of the client. Usually code can get it from backend (in this case cherrypy) or upstream (which is not feasible to do here since it will required refactoring of both WMCore and CRABServer to pass around this info. That's why I left it empty to avoid dependency on cherrypy. I bet no-one in CMS really is looking into this therefore it is fine to keep it empty. The commend just documents this fact.